src/createServer.js

```
@@ -1,8 +1,203 @@
+        amount: String(expense.amount),
+      };
+
+      try {
+        mkdirSync(path.dirname(file), { recursive: true });
+      } catch (mkdirErr) {
+        console.error('Directory create error:', mkdirErr);
+        res.statusCode = 500;
+        res.setHeader('Content-Type', 'text/plain');
+        res.end('Internal Server Error');
+
+        return;
+      }
+
+      const writer = createWriteStream(file);
+
+      writer.on('error', (error) => {
+        console.error('Write stream error:', error);
+        res.statusCode = 500;
+        res.setHeader('Content-Type', 'text/plain');
+        res.end('Internal Server Error');
+      });
+
+      const pretty = JSON.stringify(normalizedExpense, null, 2);
+
+      writer.end(pretty, () => {
+        res.statusCode = 200;
+
+        if (contentType.includes('application/json')) {
+          res.setHeader('Content-Type', 'application/json');
+          res.end(JSON.stringify(normalizedExpense));
+        } else {
+          const escaped = pretty
+            .replace(/&/g, '&amp;')
+            .replace(/</g, '&lt;')
+            .replace(/>/g, '&gt;');
+
+          res.setHeader('Content-Type', 'text/html; charset=utf-8');
+
+          res.end(
+            `<!doctype html><html><head><meta charset="utf-8"><title>Saved</title></head><body><h1>Expense saved</h1><pre>${escaped}</pre></body></html>`,
+          );
+        }
+      });
+    });
+
+    req.on('error', (err) => {
+      console.error('Request error:', err);
+      res.statusCode = 400;
+      res.setHeader('Content-Type', 'text/plain');
+      res.end('Bad Request');
+    });
+  }
+
+  server.on('request', getHandler);
+  server.on('request', postHandler);
+
+  server.on('request', (req, res) => {
+    if (req.method !== 'GET' && req.method !== 'POST') {
+      res.statusCode = 405;
+      res.setHeader('Content-Type', 'text/plain');
+      res.end('Method Not Allowed');
+    }
+  });
+
+  server.on('error', () => {});
```

The server currently registers an empty server-level error handler: `server.on('error', () => {});` which swallows errors. Replace it with a logger so server errors don't get hidden, e.g. `server.on('error', (err) => console.error('Server error:', err));` (line 198).